app/lib/feed_manager.rb

```diff
@@ -188,7 +194,7 @@ def add_to_feed(timeline_type, account, status)
   # do so if appropriate.
   def remove_from_feed(timeline_type, account, status)
     timeline_key = key(timeline_type, account.id)
-    reblog_key = key(timeline_type, account.id, 'reblogs')
+    reblog_key   = key(timeline_type, account.id, 'reblogs')
 
     if status.reblog?
```

Comment:
We should handle the case where `status.reblog? == false` and unconditionally delete the reblog_set_key, I think? If a status is deleted, we can safely toss all reblogs of it. (I don't actually know what happens right now in the case where we only have a reblog in our feed and the original is deleted, but it might be worth looking into.)

Reply:
True